### ISSUE TITLE
macOS background color won't get set unless someone resizes the window

### ISSUE THREAD

**Issue by MurmursOnSARS** [AryToNeX/Glasscord#6]
# UPDATED, GO TO THIS COMMENT: https://github.com/AryToNeX/Glasscord/issues/6#issuecomment-611365923

Old issue report (not really relevant now)
------
https://github.com/AryToNeX/Glasscord/blob/752b30500cb5a31e4bf7c8c66466fdfc847f6a1a/glasscord.js#L162-L168

This fix can cause flickering, especially with scrollbars. Currently, if someone uses the CSS editor with auto-update turned on, the effect may be harsher than epilepsy guidelines permit. Switching from `width` to `height` helps, but there should be a nicer way to update the window.

**Desktop (please complete the following information):**
 - OS: MacOS 10.15
 - Discord version: 0.0.256
 - Glasscord version: idk
 - Other Discord mods used: BandagedBD

**Comment by AryToNeX** [comment]
I am sorry to hear this! Sure we could switch width and height, but I don't know of any other method to update the window. If you do, please let me know!

**Comment by AryToNeX** [comment]
I wonder if just calling setBounds once with no changes can effectively update the window.

**Comment by AryToNeX** [comment]
To fix this we need a method to update the window that isn't setBounds.
I can change the modified bound from `width` to `height` but I cannot work around the problem otherwise (because Electron itself is bugged)

**Comment by oofsauce** [comment]
Can't reproduce flicker on my machine.
- OS: MacOS 10.12.4
- Discord version: 0.0.256
- Glasscord version: latest
- Other Discord mods used: EnhancedDiscord & BandagedBD

**Comment by AryToNeX** [comment]
We need to shift the point of view of this issue to find a proper fix, I see

**Comment by AryToNeX** [comment]
Let's go through the issue once again, but considering everything we know currently.

In the Discord Support Server, some users noticed that BrowserWindow#setBackgroundColor() won't work on macOS unless someone resized the window manually. Then, it seems that the window updates itself and when doing so it also applies the new requested color.

Knowing this fact, I implemented a workaround to the issue: I instructed Glasscord to resize the window to make it one pixel larger and then back to the previous size. This worked as expected and did indeed fix the problem for many users.

However, it seems that under some common circumstances (using the CSS live editor on BandagedBD) the window begins to flicker everytime the CSS is edited, that's because Glasscord tries to update the window with that hack upon a CSS update, which in theory should keep things consistent, but in practice it also triggers epilepsy to sensitive people who type moderately fast on the editor; plus, it's objectively ugly to see.

Yesterday I came up with another workaround to the issue caused by my previous workaround: we will cache the background color internally so that if the same color gets requested twice, the request will be discarded. This fixes the constant flickering we had before, but it doesn't fix it when themers want to modify the --glasscord-tint variable dynamically.

We also are totally sure this is an Electron bug and not a Glasscord bug to begin with. Our goal here is to find an unobtrusive way to still 'update' the window without causing any major issues like flickering or the user noticing.
